tests/models.py

```diff
@@ -289,6 +289,48 @@ def jinja_udf(name):
         result = task.render_template('', "{{ 'world' | hello}}", dict())
         self.assertEqual(result, 'Hello world')
 
+    def test_descendants(self):
```

Comment:
Could you specify what you mean by non happy flow tests (I get the extra check on making sure not unrelated tasks are cleared though)?